content/blog/2019-01-28-subquery-support.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Subquery Support
+created_at: 2019-01-28
+kind: article
+author_name: Ganesh Vernekar
+---
+
+## Introduction
+
+As the title suggests, a subquery is a part of a query, and allows you to do a range query within a query, which was not possible before. It has been a long-standing feature request: [prometheus/prometheus/1227](https://github.com/prometheus/prometheus/issues/1227).
+
+The [pull request](https://github.com/prometheus/prometheus/pull/4831) for subquery support was recently merged into Prometheus and will be available in Prometheus 2.7. Let’s learn more about it below.
+
+## Motivation
+
+Previously if you wanted to have a range selection on a query for your alerting rules or graphing, it would require you to have a recording rule based on that query, and perform range selection on the metrics created by the recording rules. Example: `max_over_time(rate(my_counter_total[5m])[1h])`.
```

Comment:
I would add something like this:

```suggestion

Sometimes, there are cases when you want to spot a problem using `rate` with lower resolution/interval (e.g `1m`), while aggregating this data for higher resolution (e.g `max_over_time` for `1h`).

Previously, the above was not possible for single *promQL* query. If you wanted to have a range selection on a query for your alerting rules or graphing, it would require you to have a recording rule based on that query, and perform range selection on the metrics created by the recording rules. Example: `max_over_time(rate(my_counter_total[5m])[1h])`.
```